File: esportskicentar/src/main/java/tech/esc/esportskicentar/model/Administrator.java
package tech.esc.esportskicentar.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;
import tech.esc.esportskicentar.annotation.ValidPhoneNumber;

@Entity
@Data
@EqualsAndHashCode(callSuper = false)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Administrator {
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Id
    @Column(name = "id_administrator")
    private int idAdministrator;
    @NotBlank
    @Basic
    @Column(name = "ime")
    private String ime;
    @NotBlank
    @Basic
    @Column(name = "prezime")
    private String prezime;
    @NotNull
    @ValidPhoneNumber
    @Basic
    @Column(name = "broj_telefona")
    private String brojTelefona;
    @NotBlank
    @Basic
    @Column(name = "korisnicko_ime")
    private String korisnickoIme;
    @NotBlank
    @Basic
    @Column(name = "lozinka")
    private String lozinka;
    @NotNull
    @Email
    @Basic
    @Column(name = "email")
    private String email;


}

File: esportskicentarapp/src/app/models/administrator.ts
export class Administrator {
    idAdministrator!: number; 
    ime: string | null;
    prezime: string | null; 
    brojTelefona: string | null; 
    korisnickoIme: string | null; 
    lozinka: string | null; 
    email: string | null; 

    constructor(ime: string | null, prezime: string | null, brojTelefona: string | null, 
        korisnickoIme: string | null, lozinka: string | null, email: string | null) {
        this.ime = ime; 
        this.prezime = prezime; 
        this.brojTelefona = brojTelefona; 
        this.korisnickoIme = korisnickoIme; 
        this.lozinka = lozinka; 
        this.email = email; 
    }
}